python/GafferUI/Viewer.py

```diff
@@ -240,6 +240,11 @@ def __keyPress( self, widget, event ) :
 						t["active"].setValue( False )
 				return True
 
+			if event.key == "O" and t["active"].getValue() and "orientation" in t :
+				o = t["orientation"].getValue()
+				numPresets = len( [ v for v in Gaffer.Metadata.registeredValues( t["orientation"] ) if v.startswith( "preset:" ) ] )
+				t["orientation"].setValue( ( o + 1 ) % numPresets )
```

Comment:
This would be more robust and general if it used the NodeAlgo API which abstracts away the metadata conventions for presets, and supports additional ways of specifying them. So we'd use `NodeAlgo.presets()` to get the list of presets, `currentPreset()` to find the index of the current preset in that list, and then `applyPreset()` to do the increment.

Reply:
Ah, very nice, I didn't know about those `NodeAlgo` methods. Fixed in https://github.com/GafferHQ/gaffer/pull/5207/commits/c6b8f793d94feea1526781c49fce9a56c172971c